### DISCUSSION TITLE
Nuxt 3: When useFetch returns an error, how can I get the response's status code both in SSR and on the client?

### DISCUSSION THREAD

**Discussion by ataylor32** [nuxt/nuxt#17864] [Questions] [answered]
Suppose I have a route for viewing a blog post. In that route's .vue file, I call `useFetch` to fetch the blog post. If I fetch a non-existent blog post, I want to display a custom "not found" message. But if there is, say, a 500 error, I want to display a different message. So I need access to the response's status code. This is what I'm doing now:

```typescript
blogPostStatus.value = 'loading';

const { data, error } = await useFetch(...);

if (data.value) {
  blogPost.value = data.value as BlogPostInterface;
  blogPostStatus.value = 'loaded';
} else if (error.value.response?.status === 404) {
  blogPostStatus.value = 'not_found';

  if (nuxtApp.ssrContext) {
    nuxtApp.ssrContext.nuxt.error = createError({ statusCode: 404 });
  }
} else {
  blogPostStatus.value = 'error';
}
```

The 404 handling is working as intended in SSR (the `else if (error.value.response?.status === 404` block is what gets run). But then when the code runs on the client, `error.value` is `true`, so the `else` block is what gets run. Thus, on the server `blogPostStatus.value` gets set to `'not_found'` but on the client, it gets set to `'error'` and there is a hydration text mismatch. What is the correct way to handle this?

**Comment by danielroe** [comment, marked as answer]
This is the issue to follow: https://github.com/nuxt/framework/issues/2122
